driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java

```diff
@@ -79,7 +81,21 @@ public TLSSocketChannel( ByteChannel channel, Logger logger, SSLEngine sslEngine
         this.cipherIn = ByteBuffer.allocate( sslEngine.getSession().getPacketBufferSize() );
         this.plainOut = ByteBuffer.allocate( sslEngine.getSession().getApplicationBufferSize() );
         this.cipherOut = ByteBuffer.allocate( sslEngine.getSession().getPacketBufferSize() );
-        runHandshake();
+
+        try
+        {
+            runHandshake();
+        }
+        catch ( SSLHandshakeException e )
+        {
+            Throwable error = e;
+            if( error.getCause() != null ) // get the real exception
+            {
+                error = e.getCause();
+            }
+            throw new ClientException( "Failed to establish secured connection with the server: " + error.getMessage(),
```

Comment:
It is probably better to move this `try-catch` block inside `runHandshake()` and just re-throw `ClientException` with whatever `IOException` is thrown as the cause.